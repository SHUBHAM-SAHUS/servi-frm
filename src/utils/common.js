import {
  JWT_ID_TOKEN,
  JWT_ACCESS_TOKEN,
  JWT_REFRESH_TOKEN,
  USER_NAME,
  ADMIN_DETAILS,
  PAYMENT_DETAILS,
  ACCESS_CONTROL
} from '../dataProvider/constant';
import $ from 'jquery';
import { browserHistory } from '../components/Routes/headRoute';

export const goBackBrowser = (props, newPath) => {
  let pathname = props.history.location.pathname;
  if (pathname === 'signin' || pathname === 'signin_code' || pathname === 'signup') {
    return;
  } else {
    browserHistory.goBack()
  }
}

export const goBack = (props, newPath) => {
  let pathname = props.history.location.pathname;
  if (pathname === 'signin' || pathname === 'signin_code' || pathname === 'signup') {
    return;
  } else {
    window.location.replace('/dashboard/dashboard')

  }
}

export const abbrivationStr = (text) => {
  let array = text.split(" ", 2)
  if (array.length > 1) {
    let abbr = array[0].charAt(0) + array[1].charAt(0)

    return abbr.toUpperCase();
  } else {
    let abbr = array[0].charAt(0) + array[0].charAt(1)
    return abbr.toUpperCase();
  }
}

/* 
Allow ph no +61nxxxxxxxx or 0nxxxxxxxx where n can be 2, 3, 4, 7, and 
This is 12 chars including +, with the 4th char in [2, 3, 4, 7, 8] 
or 10 chars with the 2nd char in [2, 3, 4, 7, 8].
*/
export const customPhonoeNoValidate = (phone_number) => {
  var substr = ''
  if ((phone_number.startsWith("+61") && phone_number.length === 12)) {
    substr = phone_number.substring(3, 4)
  }

  if ((phone_number.startsWith("0") && phone_number.length === 10)) {
    substr = phone_number.substring(1, 2)
  }

  if ((substr == 2 || substr == 3 || substr == 4 || substr == 7 || substr == 8)) {
    return true //allow phone no
  } else {
    return false //show error
  }
}

export const getStorage = key => {
  // return localStorage.getItem(key);
  return sessionStorage.getItem(key);
}

export const setStorage = (key, value) => {
  // return localStorage.setItem(key, value);
  return sessionStorage.setItem(key, value);
}

export const authRedirect = (history) => {
  if (getStorage(JWT_ID_TOKEN) && getStorage(JWT_ACCESS_TOKEN) && getStorage(JWT_REFRESH_TOKEN)
    && getStorage(USER_NAME) && getStorage(ADMIN_DETAILS) && getStorage(PAYMENT_DETAILS)
    && getStorage(ACCESS_CONTROL)) {
    history.push('/dashboard');
  }
}

export function groupBy(list, keyGetter) {
  const map = new Map();
  list.forEach((item) => {
    const key = keyGetter(item);
    const collection = map.get(key);
    if (!collection) {
      map.set(key, [item]);
    } else {
      collection.push(item);
    }
  });
  return map;
}

export const map_to_obj = (aMap => {
  const obj = {};
  aMap.forEach((v, k) => { obj[k] = v });
  return obj;
});

export const countries = ["Afghanistan", "Albania", "Algeria", "American Samoa", "Angola", "Anguilla", "Antigua and Barbuda", "Argentina", "Armenia", "Aruba", "Australia", "Austria", "Azerbaijan", "Bahamas", "Bahrain", "Bangladesh", "Barbados", "Belarus", "Belgium", "Belize", "Benin", "Bermuda", "Bhutan", "Bolivia", "Bonaire, Saint Eustatius and Saba (Netherlands Carribean)", "Bosnia and Herzegovina", "Botswana", "Brazil", "British Indian Ocean Territory", "Brunei Darussalam", "Bulgaria", "Burkina Faso", "Burundi", "Cambodia", "Cameroon", "Canada", "Cape Verde", "Cayman Islands", "Central African Republic", "Chad", "Chile", "China", "Colombia", "Comoros", "Congo (Democratic Republic)", "Congo (Republic)", "Cook Islands", "Costa Rica", "Cote d'Ivoire", "Croatia", "Cuba", "Cura√ßao", "Cyprus", "Czech Republic", "Denmark", "Djibouti", "Dominica", "Dominican Republic", "Ecuador", "Egypt", "El Salvador", "Equatorial Guinea", "Eritrea", "Estonia", "eSwatini (formerly Swaziland)", "Ethiopia", "Falkland Islands (Malvinas)", "Faroe Islands", "Fiji", "Finland", "France", "French Guiana (Guyana)", "French Polynesia", "Gabon", "Gambia", "Georgia", "Germany", "Ghana", "Gibraltar", "Greece", "Greenland", "Grenada", "Guadeloupe", "Guam", "Guatemala", "Guinea", "Guinea-Bissau", "Guyana", "Haiti", "Honduras", "Hong Kong", "Hungary", "Iceland", "India", "Indonesia", "Iran", "Iraq", "Ireland", "Israel", "Italy", "Jamaica", "Japan", "Jordan", "Kazakhstan", "Kenya", "Kribati", "Kosovo", "Kuwait", "Kyrgyzstan", "Laos", "Latvia", "Lebanon", "Lesotho", "Libya", "Liberia", "Liechtenstein", "Lithuania", "Luxemborg", "Macao", "Madagascar", "Malawi", "Malaysia", "Maldives", "Mali", "Malta", "Marshall Islands", "Martinique", "Mauritania", "Mauritius", "Mexico", "Micronesia", "Moldova", "Mongolia", "Montenegro", "Montserrat", "Morocco", "Mozambique", "Myanmar (Burma)", "Namibia", "Nauru", "Nepal", "Netherlands", "New Caledonia", "New Zealand", "Nicaragua", "Niger", "Nigeria", "Niue", "North Korea", "North Macedonia (formerly Macedonia)", "North Mariana Islands", "Norway", "Oman", "Pakistan", "Palau", "Panama", "Papua New Guinea", "Paraguay", "Peru", "Philippines", "Pitcairn", "Poland", "Portugal", "Puerto Rico", "Qatar", "Reunion", "Romania", "Russia", "Rwanda", "Saint Helena, Ascension and Tristan da Cunha", "Saint Kitts and Nevis", "Saint Martin (French Part)", "Saint Lucia", "Saint Pierre and Miquelon", "Saint Vincent and the Grenadines", "Samoa", "Sao Tome and Principe", "Saudi Arabia", "Senegal", "Serbia", "Seychelles", "Sierra Leone", "Singapore", "Slovakia", "Slovenia", "Soloman Islands", "Somalia", "South Africa", "South Korea", "Spain", "Sri Lanka", "Sudan", "Suriname", "Sweden", "Switzerland", "Syria", "Taiwan", "Tajikstan", "Tanzania", "Timor Leste", "Togo", "Tokelau", "Tonga", "Trinidad and Tobago", "Tunisia", "Turkey", "Turkmenistan", "Turks and Caicos Islands", "Tuvalu", "Uganda", "Ukraine", "United Arab Emirates", "United Kingdom", "United States", "United States Minor Outlying Islands", "Uruguay", "Uzbekistan", "Vanuatu", "Virgin Islands (British)", "Virgin Islands (US)", "Vatican City", "Venezuela", "Vietnam", "Wallis and Futuna", "Yemen", "Zambia", "Zimbabwe"];

export const zones = ["Perth", "Western Australia", "Northern Territory", "Darwin", "Queensland", "South Australia", "New South Wales", "Adelaide", "Victoria", "Melbourne", "Tasmania", "Hobart", "Canberra", "Sydney", "Brisbane"];

export const days = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'];

export const monthNames = ["Jan", "Feb", "Mar", "Apr", "May", "June",
  "July", "Aug", "Sept", "Oct", "Nov", "Dec"
];

export const users = [
  {
    name: "Anurag",
    id: 1
  },
  {
    name: "Kalpesh",
    id: 2
  },
  {
    name: "Arish",
    id: 3
  },
  {
    name: "Rupesh",
    id: 4
  },
  {
    name: "Sumeet",
    id: 5
  },
  {
    name: "Ramanath",
    id: 6
  },
  {
    name: "Prasad",
    id: 7
  },
  {
    name: "Vishnu",
    id: 8
  },
  {
    name: "Vishal",
    id: 9
  },
  {
    name: "Samruddhi",
    id: 10
  }
]

export const incidents = [
  {
    name: "Injury",
    id: 1,
  },
  {
    name: "Environment",
    id: 2
  },
  {
    name: "Theft",
    id: 3
  },
  {
    name: "Plant",
    id: 4
  },
  {
    name: "Process",
    id: 5
  },
  {
    name: "Property",
    id: 6
  },
  {
    name: "Drug/Alcohol",
    id: 7
  },
  {
    name: "Security",
    id: 8
  },
  {
    name: "Near Miss",
    id: 9
  },
  {
    name: "Other",
    id: 10
  }
]

export const states = [
  {
    name: "Victoria",
    id: 1
  },
  {
    name: "Tasmania",
    id: 2
  },
  {
    name: "Queensland",
    id: 3
  },
  {
    name: "Northern Territory",
    id: 4
  },
  {
    name: "Maharashtra",
    id: 5
  }
]

export function modifyObject(obj, keyPath, value) {
  let lastKeyIndex = keyPath.length - 1;
  for (var i = 0; i < lastKeyIndex; ++i) {
    let key = keyPath[i];
    if (!(key in obj)) {
      obj[key] = {}
    }
    obj = obj[key];
  }
  obj[keyPath[lastKeyIndex]] = value;
}

export function removeByAttr(arr, attr, value) {
  var i = arr.length;
  while (i--) {
    if (arr[i]
      && arr[i].hasOwnProperty(attr)
      && (arguments.length > 2 && arr[i][attr] === value)) {

      arr.splice(i, 1);

    }
  }
  return arr;
}

export function DeepTrim(obj) {
  for (var prop in obj) {
    var value = obj[prop], type = typeof value;
    if (value != null && (type == "string" || type == "object") && obj.hasOwnProperty(prop)) {
      if (type == "object") {
        obj[prop] = DeepTrim(obj[prop]);
      } else {
        obj[prop] = obj[prop].trim();
      }
    }
  }
  return obj;
}

export function currencyFormat(value) {
  var formatedCurrency = new Intl.NumberFormat('en-AU', { style: 'currency', currency: 'aud' }).format(value)
  return formatedCurrency
}

export const handleFocus = (errors, str) => {
  for (let key in errors) {
    if (typeof errors[key] == 'string') {
      $(str + key + "-focus").focus();
      return str + key;
    } else if (Array.isArray(errors[key]) && errors[key].length > 0) {
      for (let i = 0; i < errors[key].length; i++) {
        if (errors[key][i])
          if (typeof handleFocus(errors[key][i], str + key + "\\[" + i + "\\]\\.") == 'string')
            return handleFocus(errors[key][i], str + key + "\\[" + i + "\\]\\.");
      }
    } else if (typeof errors[key] == "object" && Object.keys(errors[key]).length > 0) {
      if (typeof handleFocus(errors[key], str + key + "\\.") == 'string')
        return handleFocus(errors[key], str + key + "\\.");
    }
  }

}

export const calculateRisk = (likelihood_control, consequence_control) => {
  if (likelihood_control && consequence_control) {
    let key_before_control = "$" + likelihood_control.order + "_" + consequence_control.order
    let resultObj = {
      $1_1: "1. Low",
      $1_2: "2. Low",
      $1_3: "3. Low",
      $1_4: "4. Moderate",
      $1_5: "5. Moderate",

      $2_1: "2. Low",
      $2_2: "4. Moderate",
      $2_3: "6. Moderate",
      $2_4: "8. High",
      $2_5: "10. High",

      $3_1: "3. Low",
      $3_2: "6. Moderate",
      $3_3: "9. High",
      $3_4: "12. Very High",
      $3_5: "15. Very High",

      $4_1: "4. Moderate",
      $4_2: "8. high",
      $4_3: "12. Very High",
      $4_4: "16. Extreme",
      $4_5: "20. Extreme",

      $5_1: "5. Moderate",
      $5_2: "10. High",
      $5_3: "15. Very High",
      $5_4: "20. Extreme",
      $5_5: "25. Extreme"
    }
    return resultObj[key_before_control];

  }
  /* let key_before_control = (likelihood_control_name && consequence_control_name) ? likelihood_control_name.replace(/ /g, "_") + "_" + consequence_control_name.replace(/ /g, "_") : null;

  let resultObj = {
    Almost_Certain_Insignificant: 'Medium',
    Almost_Certain_Minor: 'High',
    Almost_Certain_Moderate: 'High',
    Almost_Certain_Major: 'Extreme',
    Almost_Certain_Severe: 'Extreme',

    Likely_Insignificant: 'Medium',
    Likely_Minor: 'Medium',
    Likely_Moderate: 'High',
    Likely_Major: 'Extreme',
    Likely_Severe: 'Extreme',

    Possible_Insignificant: 'Medium',
    Possible_Minor: 'Medium',
    Possible_Moderate: 'High',
    Possible_Major: 'High',
    Possible_Severe: 'Extreme',

    Unlikely_Insignificant: 'Low',
    Unlikely_Minor: 'Medium',
    Unlikely_Moderate: 'Medium',
    Unlikely_Major: 'High',
    Unlikely_Severe: 'High',

    Rare_Insignificant: 'Low',
    Rare_Minor: 'Low',
    Rare_Moderate: 'Medium',
    Rare_Major: 'High',
    Rare_Severe: 'High',
  }

  return resultObj[key_before_control]; */
}

export const showAsCSV = stringList => {
  return (Array.isArray(stringList) && stringList.length > 0) ? stringList.reduce(((acc, cVal) => acc + ', ' + cVal.area_name), ',').split(',,')[1] : '-'
}

export const resolveDurationToNumber = duration => {
  return parseInt(duration.split('_')[0])
}