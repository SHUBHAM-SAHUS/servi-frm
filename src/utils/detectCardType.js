export const detectCardType = (number) => {
    if (number) {
        var re = new RegExp("^4");
        if (number.match(re) != null) {
            return "Visa";
        }
        if (number.match(/^5[1-5]/) && number.match(/^\d{1,16}$/)) {
            return 'mastercard';
        }
        if (number.match(/^3[47]/) && (number.match(/^\d{1,15}$/))) {
            return 'amex';
        }

        if (number.match(/^3(?:0[0-5]|[68][0-9])[0-9]{4,}$/)) {
            return 'dinner';
        }
    }
}