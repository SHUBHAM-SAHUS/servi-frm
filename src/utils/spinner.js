import { STOP_SPINNER, START_SPINNER, STOP_MINI_SPINNER, START_MINI_SPINNER, STOP_SCROLL_SPINNER, START_SCROLL_SPINNER } from '../dataProvider/constant'

export const stopSipnner = (dispatch) => {
  dispatch({
    type: STOP_SPINNER
  });
}

export const startSipnner = (dispatch) => {
  dispatch({
    type: START_SPINNER
  });
}

export const stopMiniSpinner = dispatch => {
  dispatch({
    type: STOP_MINI_SPINNER
  });
}

export const startMiniSpinner = (dispatch) => {
  dispatch({
    type: START_MINI_SPINNER
  });
}

export const startScrollSpinner = dispatch => {
  dispatch({
    type: START_SCROLL_SPINNER
  })
}
export const stopScrollSpinner = dispatch => {
  dispatch({
    type: STOP_SCROLL_SPINNER
  })
}