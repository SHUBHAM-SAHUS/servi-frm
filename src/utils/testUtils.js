//import checkPropTypes from 'check-prop-types';
import { applyMiddleware, createStore } from 'redux';
import reducers from '../reducers';
import { middlewares } from './createStore';

export const findByClassName = (component, attr) => {
    const wrapper = component.find(`.${attr}`);
    return wrapper;
};

/* export const checkProps = (component, expectedProps) => {
    const propsErr = checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
    return propsErr;
}; */

export const testStore = (initialState) => {
    const createStoreWithMiddleware = applyMiddleware(...middlewares)(createStore);
    return createStoreWithMiddleware(reducers, initialState);
};