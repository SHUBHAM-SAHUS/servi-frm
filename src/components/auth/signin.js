import React from 'react';
import { reduxForm, Field } from 'redux-form';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { Link } from 'react-router-dom';
import { notification } from 'antd';
import { Strings } from '../../dataProvider/localize'
import * as actions from '../../actions';
import { customInput } from '../common/custom-input';
import { validate } from '../../utils/validations';
import GoogleCaptcha from '../common/GoogleCaptcha';
import {
	MFA_STA,
	SET_PWD,
	JWT_SESSION_TOKEN,
	LOGIN,
	JWT_ID_TOKEN,
	JWT_ACCESS_TOKEN,
	USER_NAME,
	REMEMBER_ME
} from '../../dataProvider/constant';
import { getStorage, setStorage, authRedirect, handleFocus, DeepTrim } from '../../utils/common';

class SignIn extends React.Component {

	constructor(props) {
		super(props);
		this.state = {
			password: '',
			count: 0,
			fromChild: '',
			rememberMeChecked: false
		}
		this.passwordStrength = this.passwordStrength.bind(this);
		this.handleData = this.handleData.bind(this);
	}

	componentDidMount() {
		authRedirect(this.props.history);
		this.props.resetReduxStore();
	}

	onSubmit = async (formData) => {
		formData = await DeepTrim(formData);

		this.props.signIn(formData)
			.then((res) => {
				this.props.cleanErrorMsg()
				if (this.props.loginStatus) {
					setStorage(REMEMBER_ME, this.state.rememberMeChecked)
				}
				if (this.props.loginStatus && this.props.authType === MFA_STA && getStorage(JWT_SESSION_TOKEN)) {
					this.props.history.push('/signin_code');
				}
				else if (this.props.loginStatus && this.props.authType === SET_PWD && getStorage(JWT_SESSION_TOKEN)) {
					this.props.history.push('/signup');
				}
				else if (this.props.loginStatus && this.props.authType === LOGIN && getStorage(JWT_ID_TOKEN) && getStorage(JWT_ACCESS_TOKEN)) {
					this.props.history.push('/dashboard');
				}
			})
			.catch((message) => {
				if (this.props.errorMessage) {
					this.setState({
						count: this.state.count + 1
					})
				} else {
					return notification.error({
						message: Strings.error_title,
						description: message ? message : Strings.generic_error, onClick: () => { },
						className: 'ant-error'
					});
				}
			});
	}

	passwordStrength(e) {
		this.setState({
			password: e.target.value
		});
	}

	handleData(data) {
		this.setState({
			fromChild: data
		});
	}

	handleRememberMe = (event) => {
		this.setState({ rememberMeChecked: event.target.checked })
	}

	handleNotMeClick = () => {
		// localStorage.removeItem(USER_NAME);
		sessionStorage.removeItem(USER_NAME);
		this.setState({})
	}

	render() {
		const { handleSubmit } = this.props;
		return (
			<div>
				<div className="sf-login">
					<img className="sf-logo" src="../images/service_form_big.png" alt="SF logo" />
					<h2 className="sf-lg-heading">{Strings.sign_in_title_login}</h2>
					<form onSubmit={handleSubmit(this.onSubmit)}>
						<fieldset className="userTxtbox fldset-box outr-notme">
							<svg className="svg-ico-f" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><path d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z" /><path d="M0 0h24v24H0z" fill="none" /></svg>
							<Field
								name="email_address"
								type="text"
								id="email_address"
								label={Strings.sign_in_email_label}
								placeholder={Strings.sign_in_email_placeholder}
								readonly={getStorage(USER_NAME) ? true : false}
								component={customInput} />
							{getStorage(USER_NAME) ? <button className="notmetxt" onClick={this.handleNotMeClick} type="button">Not me</button> : null}
						</fieldset>

						<fieldset className="passTxtbox fldset-box">
							<svg className="svg-ico-f" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><path d="M0 0h24v24H0z" fill="none" /><path d="M12.65 10C11.83 7.67 9.61 6 7 6c-3.31 0-6 2.69-6 6s2.69 6 6 6c2.61 0 4.83-1.67 5.65-4H17v4h4v-4h2v-4H12.65zM7 14c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z" /></svg>
							<Field
								name="password"
								type="password"
								id="password"
								placeholder={Strings.sign_in_password_label}
								label={Strings.sign_in_password_placeholder}
								onChange={this.passwordStrength}
								component={customInput} />
						</fieldset>


						{this.props.errorMessage ?
							<div className="alert alert-danger">
								{this.props.errorMessage}
							</div> : null}
						{this.state.count > 2 ?
							<div className="sf-g-captcah">
								<GoogleCaptcha handlerFromParant={this.handleData} />
							</div> :
							null}
						{this.state.count > 2 && this.state.fromChild !== "Verified" ? <button type="submit" className="sf-btn" disabled>{Strings.sign_in_btn_sign_in}</button> : <button type="submit" className="sf-btn">{Strings.sign_in_btn_sign_in}</button>}
						{this.state.count > 2 ?
							<div className="forgt-pass">{Strings.sign_in_forgot_password} <Link onClick={() => this.props.cleanErrorMsg()}
								key='forgotpass' to='/forgotpass'>{Strings.sign_in_password}</Link> | <Link
									onClick={() => this.props.cleanErrorMsg()}
									key='forgotusername' to='/Forgotusername'>{Strings.sign_in_forgot_username}</Link>
							</div> : null}

						<div className="rember-me">
							<input className="styled-checkbox" id="styled-checkbox-1" type="checkbox" value="value1" onChange={this.handleRememberMe} />
							<label for="styled-checkbox-1">{Strings.sign_in_remember_me}</label>
						</div>

					</form>
				</div>
			</div>

		)
	}
}

const mapStateToProps = (state) => {
	if (getStorage(USER_NAME)) {
		var initialValues = { user_name: getStorage(USER_NAME) };
	}
	return {
		errorMessage: state.auth.errorMessage,
		authType: state.auth.authType,
		loginStatus: state.auth.loginStatus,
		initialValues: initialValues ? initialValues : {}
	}
}

export const SignInComponent = connect(mapStateToProps, actions)(SignIn);
export default compose(
	connect(mapStateToProps, actions),
	reduxForm({
		form: 'signin', validate,
		onSubmitFail: (errors, dispatch, sub, props) => {
			handleFocus(errors, "#");
		}
	})
)(SignIn)