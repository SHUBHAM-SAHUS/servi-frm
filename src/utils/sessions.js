import {
	JWT_ID_TOKEN,
	JWT_ACCESS_TOKEN,
	JWT_REFRESH_TOKEN,
	USER_NAME,
	ADMIN_DETAILS,
	FIRST_TIME_LOGIN,
	PAYMENT_DETAILS,
	ACCESS_CONTROL, REMEMBER_ME,
	ORGANIZATIONS_LIST,
	SELECTED_ORG
} from '../dataProvider/constant'
import { getStorage, setStorage } from './common';
import { getCurrentOrganization } from '../actions/organizationAction';
import { Store } from '../index';

export const storeLoginData = (res, selectedOrg = 0) => {
	setStorage(JWT_ID_TOKEN, res.data.data.auth_data.id_token);
	setStorage(JWT_ACCESS_TOKEN, res.data.data.auth_data.access_token);
	setStorage(JWT_REFRESH_TOKEN, res.data.data.auth_data.refresh_token);
	setStorage(USER_NAME, res.data.data.admin_details.user_name);
	setStorage(ADMIN_DETAILS, JSON.stringify({
		...res.data.data.admin_details, organisation: res.data.data.organizations[selectedOrg],
		role: res.data.data.organizations[selectedOrg].role
	}));
	setStorage(FIRST_TIME_LOGIN, res.data.data.admin_details.first_time_login);
	setStorage(PAYMENT_DETAILS, JSON.stringify({
		subscription_amount: res.data.data.organizations[selectedOrg].subscription_amount,
		is_payment_due: res.data.data.organizations[selectedOrg].is_payment_due,

	}));
	setStorage(ACCESS_CONTROL,/*  JSON.stringify(accessControl.access_controls) */JSON.stringify(res.data.data.organizations[selectedOrg].access_controls));
	setStorage(ORGANIZATIONS_LIST, JSON.stringify(res.data.data.organizations));
	getCurrentOrganization(Store.dispatch);
}

export const changeOrganization = (selectedOrg = 0) => {
	setStorage(ADMIN_DETAILS, JSON.stringify({
		...JSON.parse(getStorage(ADMIN_DETAILS)), organisation: JSON.parse(getStorage(ORGANIZATIONS_LIST))[selectedOrg],
		role: JSON.parse(getStorage(ORGANIZATIONS_LIST))[selectedOrg].role
	}));
	setStorage(PAYMENT_DETAILS, JSON.stringify({
		subscription_amount: JSON.parse(getStorage(ORGANIZATIONS_LIST))[selectedOrg].subscription_amount,
		is_payment_due: JSON.parse(getStorage(ORGANIZATIONS_LIST))[selectedOrg].is_payment_due,
		eway_token: JSON.parse(getStorage(ORGANIZATIONS_LIST))[selectedOrg].eway_token,
		card_details: JSON.parse(getStorage(ORGANIZATIONS_LIST))[selectedOrg].card_details
	}));
	setStorage(ACCESS_CONTROL,/*  JSON.stringify(accessControl.access_controls) */JSON.stringify(JSON.parse(getStorage(ORGANIZATIONS_LIST))[selectedOrg].access_controls));
	getCurrentOrganization(Store.dispatch);
}

export const removeLoginData = () => {
	if (!JSON.parse(getStorage(REMEMBER_ME)))
		sessionStorage.removeItem(USER_NAME);
	sessionStorage.removeItem(JWT_ID_TOKEN);
	sessionStorage.removeItem(JWT_ACCESS_TOKEN);
	sessionStorage.removeItem(JWT_REFRESH_TOKEN);
	sessionStorage.removeItem(ADMIN_DETAILS);
	sessionStorage.removeItem(FIRST_TIME_LOGIN);
	sessionStorage.removeItem(PAYMENT_DETAILS);
	sessionStorage.removeItem(ACCESS_CONTROL);
	sessionStorage.removeItem(ORGANIZATIONS_LIST);
	sessionStorage.removeItem(SELECTED_ORG);
	/* localStorage.removeItem(USER_NAME);
localStorage.removeItem(JWT_ID_TOKEN);
localStorage.removeItem(JWT_ACCESS_TOKEN);
localStorage.removeItem(JWT_REFRESH_TOKEN);
localStorage.removeItem(ADMIN_DETAILS);
localStorage.removeItem(FIRST_TIME_LOGIN);
localStorage.removeItem(PAYMENT_DETAILS);
localStorage.removeItem(ACCESS_CONTROL);
localStorage.removeItem(ORGANIZATIONS_LIST);
localStorage.removeItem(SELECTED_ORG); */
}
