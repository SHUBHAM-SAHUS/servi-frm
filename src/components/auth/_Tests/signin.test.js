import {SignInComponent} from '../signin';
import { shallow } from 'enzyme';
import { findByClassName, testStore } from '../../../utils/testUtils';
import React from 'react';

const store = testStore();
let handleSubmit = jest.fn();
const setUp = () => {
    const wrapper = shallow(<SignInComponent store={store} handleSubmit={handleSubmit}/>).childAt(0).dive();
    return wrapper;
};

describe('SignIn Component', () => {

    let wrapper;
    beforeEach(() => {
        wrapper = setUp();
    });

    it('Should render without errors', () => {
        const component = findByClassName(wrapper, 'sf-login');
        expect(component.length).toBe(1);
    });

    it('should render correctly in SignInComponent', () => {
        const component = shallow(<SignInComponent store={store} />);
      
        expect(component).toMatchSnapshot();
      });
});