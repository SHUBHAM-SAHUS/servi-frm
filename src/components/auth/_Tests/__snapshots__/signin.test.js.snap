// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn Component should render correctly in SignInComponent 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UNDEFINED
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <SignIn
        authType=""
        changePassword={[Function]}
        cleanErrorMsg={[Function]}
        errorMessage=""
        forgetUser={[Function]}
        forgotPass={[Function]}
        initialValues={Object {}}
        loginStatus={0}
        resetPass={[Function]}
        resetReduxStore={[Function]}
        signIn={[Function]}
        signInNextStep={[Function]}
        signOut={[Function]}
        signUp={[Function]}
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
        verifyAttributes={[Function]}
      />,
      "value": null,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "authType": "",
        "changePassword": [Function],
        "cleanErrorMsg": [Function],
        "errorMessage": "",
        "forgetUser": [Function],
        "forgotPass": [Function],
        "initialValues": Object {},
        "loginStatus": 0,
        "resetPass": [Function],
        "resetReduxStore": [Function],
        "signIn": [Function],
        "signInNextStep": [Function],
        "signOut": [Function],
        "signUp": [Function],
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "verifyAttributes": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <SignIn
          authType=""
          changePassword={[Function]}
          cleanErrorMsg={[Function]}
          errorMessage=""
          forgetUser={[Function]}
          forgotPass={[Function]}
          initialValues={Object {}}
          loginStatus={0}
          resetPass={[Function]}
          resetReduxStore={[Function]}
          signIn={[Function]}
          signInNextStep={[Function]}
          signOut={[Function]}
          signUp={[Function]}
          store={
            Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            }
          }
          verifyAttributes={[Function]}
        />,
        "value": null,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "authType": "",
          "changePassword": [Function],
          "cleanErrorMsg": [Function],
          "errorMessage": "",
          "forgetUser": [Function],
          "forgotPass": [Function],
          "initialValues": Object {},
          "loginStatus": 0,
          "resetPass": [Function],
          "resetReduxStore": [Function],
          "signIn": [Function],
          "signInNextStep": [Function],
          "signOut": [Function],
          "signUp": [Function],
          "store": Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          },
          "verifyAttributes": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
